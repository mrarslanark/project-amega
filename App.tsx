import React from 'react';
import {SafeAreaView, StatusBar, StyleSheet} from 'react-native';
import Navigator from './src/navigation';
import {NavigationContainer} from '@react-navigation/native';

const App: React.FC = (): React.JSX.Element => {
  return (
    <SafeAreaView style={styles.wrapper}>
      <StatusBar barStyle={'dark-content'} />
      <NavigationContainer>
        <Navigator />
      </NavigationContainer>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  wrapper: {
    flex: 1,
  },
});

export default App;
