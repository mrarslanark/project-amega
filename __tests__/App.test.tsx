import 'react-native';
import React from 'react';
import App from '../App';

// Note: test renderer must be required after react-native.
import {render, screen} from '@testing-library/react-native';

describe('App.tsx', () => {
  it('should renders successfully', () => {
    render(<App />);
    expect(screen).toMatchSnapshot();
  });
});
