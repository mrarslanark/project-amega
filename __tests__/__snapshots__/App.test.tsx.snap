// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.tsx should renders successfully 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      [
        {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <RNSScreenNavigationContainer
      style={
        {
          "flex": 1,
          "overflow": "hidden",
        }
      }
    >
      <RNSScreen
        activityState={2}
        collapsable={false}
        gestureResponseDistance={
          {
            "bottom": -1,
            "end": -1,
            "start": -1,
            "top": -1,
          }
        }
        onGestureCancel={[Function]}
        sheetAllowedDetents="large"
        sheetCornerRadius={-1}
        sheetExpandsWhenScrolledToEdge={true}
        sheetGrabberVisible={false}
        sheetLargestUndimmedDetent="all"
        style={
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 0,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="auto"
          style={
            [
              {
                "backgroundColor": "rgb(242, 242, 242)",
                "flex": 1,
              },
              [
                {
                  "flex": 1,
                  "flexDirection": "column-reverse",
                },
                undefined,
              ],
            ]
          }
        >
          <View
            style={
              {
                "flex": 1,
              }
            }
          >
            <View
              style={
                [
                  {
                    "flex": 1,
                  },
                  undefined,
                ]
              }
            >
              <RNSScreenStack
                style={
                  {
                    "flex": 1,
                  }
                }
              >
                <RNSScreen
                  collapsable={false}
                  gestureResponseDistance={
                    {
                      "bottom": -1,
                      "end": -1,
                      "start": -1,
                      "top": -1,
                    }
                  }
                  nativeBackButtonDismissalEnabled={false}
                  onAppear={[Function]}
                  onDisappear={[Function]}
                  onDismissed={[Function]}
                  onGestureCancel={[Function]}
                  onHeaderBackButtonClicked={[Function]}
                  onNativeDismissCancelled={[Function]}
                  onTransitionProgress={[Function]}
                  onWillDisappear={[Function]}
                  replaceAnimation="push"
                  sheetAllowedDetents="large"
                  sheetCornerRadius={-1}
                  sheetExpandsWhenScrolledToEdge={true}
                  sheetGrabberVisible={false}
                  sheetLargestUndimmedDetent="all"
                  stackPresentation="push"
                  style={
                    {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                  swipeDirection="horizontal"
                >
                  <View
                    accessibilityElementsHidden={false}
                    importantForAccessibility="auto"
                    style={
                      {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      collapsable={false}
                      style={
                        [
                          {
                            "flex": 1,
                          },
                          {
                            "backgroundColor": "rgb(242, 242, 242)",
                          },
                          {
                            "backgroundColor": "white",
                          },
                        ]
                      }
                    >
                      <RCTScrollView
                        contentContainerStyle={
                          {
                            "backgroundColor": "white",
                            "rowGap": 2,
                          }
                        }
                      >
                        <View>
                          <View
                            style={
                              {
                                "margin": 18,
                                "rowGap": 8,
                              }
                            }
                          >
                            <Text
                              style={
                                {
                                  "color": "black",
                                  "fontSize": 24,
                                  "fontWeight": "bold",
                                }
                              }
                            >
                              IP Tracker
                            </Text>
                            <Text
                              style={
                                {
                                  "color": "gray",
                                  "fontSize": 16,
                                  "fontWeight": "medium",
                                }
                              }
                            >
                              Find details from an IP Address. Start by searching for an IP Address below.
                            </Text>
                          </View>
                          <View
                            style={
                              {
                                "backgroundColor": "rgba(211, 208, 203, 0.2)",
                                "padding": 32,
                                "rowGap": 12,
                              }
                            }
                          >
                            <View
                              style={
                                {
                                  "rowGap": 4,
                                }
                              }
                            >
                              <TextInput
                                keyboardType="decimal-pad"
                                onChangeText={[Function]}
                                onSubmitEditing={[Function]}
                                placeholder="Search for any IP address"
                                placeholderTextColor="gray"
                                returnKeyType="done"
                                style={
                                  {
                                    "backgroundColor": "rgba(0, 0, 0, 0.05)",
                                    "borderRadius": 12,
                                    "color": "black",
                                    "padding": 16,
                                  }
                                }
                              />
                            </View>
                            <View
                              accessibilityState={
                                {
                                  "busy": undefined,
                                  "checked": undefined,
                                  "disabled": true,
                                  "expanded": undefined,
                                  "selected": undefined,
                                }
                              }
                              accessibilityValue={
                                {
                                  "max": undefined,
                                  "min": undefined,
                                  "now": undefined,
                                  "text": undefined,
                                }
                              }
                              accessible={true}
                              collapsable={false}
                              focusable={true}
                              onClick={[Function]}
                              onResponderGrant={[Function]}
                              onResponderMove={[Function]}
                              onResponderRelease={[Function]}
                              onResponderTerminate={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                              style={
                                {
                                  "alignItems": "center",
                                  "backgroundColor": "rgba(0, 0, 0, 0.05)",
                                  "borderRadius": 12,
                                  "justifyContent": "center",
                                  "opacity": 1,
                                  "padding": 12,
                                }
                              }
                            >
                              <Text
                                style={
                                  {
                                    "color": "gray",
                                    "fontSize": 14,
                                    "fontWeight": "700",
                                  }
                                }
                              >
                                SEARCH
                              </Text>
                            </View>
                          </View>
                          <View
                            style={
                              {
                                "flex": 1,
                              }
                            }
                          >
                            <View
                              style={
                                {
                                  "flex": 1,
                                }
                              }
                            >
                              <View
                                collapsable={false}
                                handlerTag={1}
                                handlerType="PanGestureHandler"
                                onGestureHandlerEvent={[Function]}
                                onGestureHandlerStateChange={[Function]}
                                onTouchEnd={[Function]}
                                onTouchStart={[Function]}
                                style={
                                  [
                                    {
                                      "overflow": "hidden",
                                    },
                                    {
                                      "height": 375,
                                      "width": 750,
                                    },
                                    {},
                                    {
                                      "flexDirection": "row",
                                    },
                                  ]
                                }
                              >
                                <View
                                  style={
                                    [
                                      {
                                        "height": 375,
                                        "position": "absolute",
                                        "width": 750,
                                      },
                                      {
                                        "transform": [
                                          {
                                            "translateX": undefined,
                                          },
                                          {
                                            "scale": undefined,
                                          },
                                        ],
                                        "zIndex": undefined,
                                      },
                                    ]
                                  }
                                  testID="__CAROUSEL_ITEM_0_NOT_READY__"
                                />
                                <View
                                  style={
                                    [
                                      {
                                        "height": 375,
                                        "position": "absolute",
                                        "width": 750,
                                      },
                                      {
                                        "transform": [
                                          {
                                            "translateX": undefined,
                                          },
                                          {
                                            "scale": undefined,
                                          },
                                        ],
                                        "zIndex": undefined,
                                      },
                                    ]
                                  }
                                  testID="__CAROUSEL_ITEM_1_NOT_READY__"
                                />
                                <View
                                  style={
                                    [
                                      {
                                        "height": 375,
                                        "position": "absolute",
                                        "width": 750,
                                      },
                                      {
                                        "transform": [
                                          {
                                            "translateX": undefined,
                                          },
                                          {
                                            "scale": undefined,
                                          },
                                        ],
                                        "zIndex": undefined,
                                      },
                                    ]
                                  }
                                  testID="__CAROUSEL_ITEM_2_NOT_READY__"
                                />
                                <View
                                  style={
                                    [
                                      {
                                        "height": 375,
                                        "position": "absolute",
                                        "width": 750,
                                      },
                                      {
                                        "transform": [
                                          {
                                            "translateX": undefined,
                                          },
                                          {
                                            "scale": undefined,
                                          },
                                        ],
                                        "zIndex": undefined,
                                      },
                                    ]
                                  }
                                  testID="__CAROUSEL_ITEM_3_NOT_READY__"
                                />
                                <View
                                  style={
                                    [
                                      {
                                        "height": 375,
                                        "position": "absolute",
                                        "width": 750,
                                      },
                                      {
                                        "transform": [
                                          {
                                            "translateX": undefined,
                                          },
                                          {
                                            "scale": undefined,
                                          },
                                        ],
                                        "zIndex": undefined,
                                      },
                                    ]
                                  }
                                  testID="__CAROUSEL_ITEM_4_NOT_READY__"
                                />
                                <View
                                  style={
                                    [
                                      {
                                        "height": 375,
                                        "position": "absolute",
                                        "width": 750,
                                      },
                                      {
                                        "transform": [
                                          {
                                            "translateX": undefined,
                                          },
                                          {
                                            "scale": undefined,
                                          },
                                        ],
                                        "zIndex": undefined,
                                      },
                                    ]
                                  }
                                  testID="__CAROUSEL_ITEM_5_NOT_READY__"
                                />
                              </View>
                            </View>
                          </View>
                        </View>
                      </RCTScrollView>
                    </View>
                  </View>
                  <RNSScreenStackHeaderConfig
                    backButtonInCustomView={false}
                    backTitleVisible={true}
                    backgroundColor="rgb(255, 255, 255)"
                    color="rgb(0, 122, 255)"
                    direction="ltr"
                    disableBackButtonMenu={false}
                    hidden={false}
                    hideBackButton={false}
                    largeTitleHideShadow={false}
                    title="Dashboard"
                    titleColor="rgb(28, 28, 30)"
                    topInsetEnabled={false}
                    translucent={false}
                  />
                </RNSScreen>
              </RNSScreenStack>
            </View>
          </View>
        </View>
      </RNSScreen>
    </RNSScreenNavigationContainer>
    <View
      collapsable={false}
      onLayout={[Function]}
      pointerEvents="auto"
      style={
        {
          "backgroundColor": "rgb(255, 255, 255)",
          "borderTopColor": "rgb(216, 216, 216)",
          "borderTopWidth": 0.5,
          "bottom": 0,
          "elevation": 8,
          "height": 49,
          "left": 0,
          "minHeight": 100,
          "paddingBottom": 0,
          "paddingHorizontal": 0,
          "position": null,
          "right": 0,
          "transform": [
            {
              "translateY": 0,
            },
          ],
        }
      }
    >
      <View
        pointerEvents="none"
        style={
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
      <View
        accessibilityRole="tablist"
        style={
          {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          accessibilityLabel="Dashboard, tab, 1 of 3"
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": true,
            }
          }
          accessibilityStates={
            [
              "selected",
            ]
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "alignItems": "center",
                "flex": 1,
              },
              {
                "backgroundColor": "transparent",
              },
              {
                "flexDirection": "column",
                "justifyContent": "flex-end",
              },
              {
                "paddingBottom": 12,
              },
            ]
          }
        >
          <View
            style={
              [
                {
                  "flex": 1,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": "100%",
                    "justifyContent": "center",
                    "minWidth": 25,
                    "position": "absolute",
                    "width": "100%",
                  },
                  {
                    "opacity": 1,
                  },
                ]
              }
            >
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  [
                    {
                      "color": "#086375",
                      "fontSize": 25,
                    },
                    undefined,
                    {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    {},
                  ]
                }
              >
                
              </Text>
            </View>
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": "100%",
                    "justifyContent": "center",
                    "minWidth": 25,
                    "position": "absolute",
                    "width": "100%",
                  },
                  {
                    "opacity": 0,
                  },
                ]
              }
            >
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  [
                    {
                      "color": "#8E8E8F",
                      "fontSize": 25,
                    },
                    undefined,
                    {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    {},
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
          <Text
            numberOfLines={1}
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "textAlign": "center",
                },
                {
                  "color": "#086375",
                },
                {
                  "fontSize": 10,
                },
                undefined,
              ]
            }
          >
            Dashboard
          </Text>
        </View>
        <View
          accessibilityLabel="Market Data, tab, 2 of 3"
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": false,
            }
          }
          accessibilityStates={[]}
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "alignItems": "center",
                "flex": 1,
              },
              {
                "backgroundColor": "transparent",
              },
              {
                "flexDirection": "column",
                "justifyContent": "flex-end",
              },
              {
                "paddingBottom": 12,
              },
            ]
          }
        >
          <View
            style={
              [
                {
                  "flex": 1,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": "100%",
                    "justifyContent": "center",
                    "minWidth": 25,
                    "position": "absolute",
                    "width": "100%",
                  },
                  {
                    "opacity": 0,
                  },
                ]
              }
            >
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  [
                    {
                      "color": "#086375",
                      "fontSize": 25,
                    },
                    undefined,
                    {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    {},
                  ]
                }
              >
                
              </Text>
            </View>
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": "100%",
                    "justifyContent": "center",
                    "minWidth": 25,
                    "position": "absolute",
                    "width": "100%",
                  },
                  {
                    "opacity": 1,
                  },
                ]
              }
            >
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  [
                    {
                      "color": "#8E8E8F",
                      "fontSize": 25,
                    },
                    undefined,
                    {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    {},
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
          <Text
            numberOfLines={1}
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "textAlign": "center",
                },
                {
                  "color": "#8E8E8F",
                },
                {
                  "fontSize": 10,
                },
                undefined,
              ]
            }
          >
            Market Data
          </Text>
        </View>
        <View
          accessibilityLabel="Profile, tab, 3 of 3"
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": false,
            }
          }
          accessibilityStates={[]}
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "alignItems": "center",
                "flex": 1,
              },
              {
                "backgroundColor": "transparent",
              },
              {
                "flexDirection": "column",
                "justifyContent": "flex-end",
              },
              {
                "paddingBottom": 12,
              },
            ]
          }
        >
          <View
            style={
              [
                {
                  "flex": 1,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": "100%",
                    "justifyContent": "center",
                    "minWidth": 25,
                    "position": "absolute",
                    "width": "100%",
                  },
                  {
                    "opacity": 0,
                  },
                ]
              }
            >
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  [
                    {
                      "color": "#086375",
                      "fontSize": 25,
                    },
                    undefined,
                    {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    {},
                  ]
                }
              >
                
              </Text>
            </View>
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": "100%",
                    "justifyContent": "center",
                    "minWidth": 25,
                    "position": "absolute",
                    "width": "100%",
                  },
                  {
                    "opacity": 1,
                  },
                ]
              }
            >
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  [
                    {
                      "color": "#8E8E8F",
                      "fontSize": 25,
                    },
                    undefined,
                    {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    {},
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
          <Text
            numberOfLines={1}
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "textAlign": "center",
                },
                {
                  "color": "#8E8E8F",
                },
                {
                  "fontSize": 10,
                },
                undefined,
              ]
            }
          >
            Profile
          </Text>
        </View>
      </View>
    </View>
  </RNCSafeAreaProvider>
</RNCSafeAreaView>
`;
